src/fides/api/util/connection_util.py

```diff
@@ -323,6 +324,26 @@ def connection_status(
 ) -> TestStatusMessage:
     """Connect, verify with a trivial query or API request, and report the status."""
 
+    # test if this is a Bigquery connection
+    if connection_config.connection_type == ConnectionType.bigquery:
```

Comment:
hmm. this will probably work, but i think there may be a cleaner way to inject the "custom" bigquery logic.

what i'd lean toward is implementing an overridden `test_connection` method in the `BigQueryConnector` class that's a subclass of the `SQLConnector` (this is the bigquery "DSR connector"): https://github.com/ethyca/fides/blob/main/src/fides/api/service/connectors/sql_connector.py#L476

that allows us to keep all the generic wrapping logic in tact, and it puts in the "custom" bigquery logic into a class specifically scoped toward bigquery. this will pave the road nicely for when we move the `BigQueryConnector` to leverage the python bigquery client, as is currently used in the d&d workflow.